src/geometry/mod.rs

```diff
@@ -1,42 +1,73 @@
 //! Geometry module.
 
 mod angle;
-mod point;
 mod real;
-mod size;
 
 pub(crate) use angle::angle_consts;
 pub(crate) use angle::Trigonometry;
 pub use angle::{Angle, AngleUnit};
-pub use point::Point;
+pub use embedded_graphics_core::geometry::{
+    AnchorPoint, Dimensions, OriginDimensions, Point, Size,
+};
 pub(crate) use real::Real;
-pub use size::Size;
-
-use crate::primitives::Rectangle;
-
-/// Adds the ability to get the dimensions/position of a graphics object
-///
-/// This **should** be implemented for all builtin embedded-graphics primitives and fonts. Third party
-/// implementations do not have to implement this trait as an object may not have a known size. If
-/// the object _does_ have a known size, this trait **should** be implemented.
-pub trait Dimensions {
-    /// Returns the bounding box.
-    fn bounding_box(&self) -> Rectangle;
+
+pub(crate) trait PointExt {
+    fn rotate_90(self) -> Self;
+
+    fn dot_product(self, other: Point) -> i32;
+
+    fn determinant(self, other: Point) -> i32;
+
+    fn length_squared(self) -> i32;
 }
 
-/// Dimensions with `top_left` of the bounding box at `(0, 0)`.
-///
-/// A blanket implementation of `Dimensions` is provided for all types that implement this trait.
-pub trait OriginDimensions {
-    /// Returns the size of the bounding box.
-    fn size(&self) -> Size;
+impl PointExt for Point {
```

Comment:
The doc comments should be moves to `PointExt`, otherwise they don't show up when using code completion.